### PR TITLE
Automatically Designing CNN

### PR DESCRIPTION
**Pull request by ThierryJudge**

### CODE REVIEW

File: deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+These functions are the learnable hyperparameters for each layer:
+
+* Number of filters: $$y_{NF} = 16x_{NF} + 16 \text{ with } x_{NF} = \{1, 2,..., 12\}$$
+* Filter height: $$y_{FH} = 2x_{FH} + 1 \text{ with } x_{FH} = \{0, 1,..., 5\}$$
+* Filter width: $$y_{FW} = 2x_{FW} + 1 \text{ with } x_{FW} = \{0, 1,..., 5\}$$
+* Pooling function: $$y_{pooling} = x_{pooling} \text{ with } x_{pooling} = \{0, 1\}$$ where 0 is maxpooling and 1 is average pooling.
+
+The segmentation network is trained with the p generated policies and a reward is obtained from each policy.
+
+To estimate the partial derivate of the policy function for each dimension (hyperparameter), each pertuation is grouped to non-overlapping categores of negative pertubations, zero pertubations and positive pertubations. $$C^d_-$$, $$C^d_0$$ and $$C^d_+$$ such that $$\pi^d_i \in \{C^d_-, C^d_0, C^d_+\}$$. This categorizations is based on the permutation that was applied in that dimension ($$\{-\epsilon^d, 0, +\epsilon^d\} $$).
```
Comment (AntoineTheb) [inline deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md]: "pertu*r*bations"

Comment (ThierryJudge) [reply deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md]: Fixed

File: deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+layout: review
+title: Automatically Designing CNN Architectures for Medical Image Segmentation
+tags: deep-learning CNN segmentation medical reinforcement cardiac
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md]: Your missing
```
author: "Thierry Judge"
```
here to honor the author of the review :wink:

File: deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    venue:   "MLMI 2018"
+pdf: "https://arxiv.org/pdf/1807.07663.pdf"
+---
+
+
+# Introduction
+
+The authors of this paper propose a new and simple technique to automatically design neural networks using policy gradients.
+The technique involves learning the optimal hyperparameters for an encoder-decoder network used for segmentation with the ACDC dataset. An optimal policy is found by iteratively permutating a random policy and updating it with the permutations that yielded the best reward (i.e. dice index).
+
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md]: You'd usually find `Dice`, first letter in capitals.

File: deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+* Number of filters: $$y_{NF} = 16x_{NF} + 16 \text{ with } x_{NF} = \{1, 2,..., 12\}$$
+* Filter height: $$y_{FH} = 2x_{FH} + 1 \text{ with } x_{FH} = \{0, 1,..., 5\}$$
+* Filter width: $$y_{FW} = 2x_{FW} + 1 \text{ with } x_{FW} = \{0, 1,..., 5\}$$
+* Pooling function: $$y_{pooling} = x_{pooling} \text{ with } x_{pooling} = \{0, 1\}$$ where 0 is maxpooling and 1 is average pooling.
+
+The segmentation network is trained with the p generated policies and a reward is obtained from each policy.
+
+To estimate the partial derivate of the policy function for each dimension (hyperparameter), each pertuation is grouped to non-overlapping categores of negative pertubations, zero pertubations and positive pertubations. $$C^d_-$$, $$C^d_0$$ and $$C^d_+$$ such that $$\pi^d_i \in \{C^d_-, C^d_0, C^d_+\}$$. This categorizations is based on the permutation that was applied in that dimension ($$\{-\epsilon^d, 0, +\epsilon^d\} $$).
+
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md]: `This categorizations is based (...)`: either `This categorization is based (...)` or `This categorizations are based (...)`

**Pull request merged**